src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -246,11 +253,33 @@ async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonS
                 "`[python-setup].resolve_all_constraints` is enabled, so "
                 "`[python-setup].requirement_constraints` must also be set."
             )
+        elif request.resolve_and_lockfile:
```

Comment:
Note that, for now, if the field is not set, we fall back to `[python-setup].experimental_lockfile`. That option will be going away soon.

Reply:
Although we are pausing the lockfile work in Pants until after PEX's lockfile support is ready to consume, I do think that landing this and following up with one more to actually delete `experimental_lockfile` would be a good place to rest on this.

Reply:
I don't know how feasible that will be: pantsbuild/pants must ignore some targets from testprojects/ when resolving. We would need an opt-out mechanism for those targets (a fake lockfile won't work because they're invalid requirements).

I'd love to finish this part of the project - I really don't think it's much work. But trying to keep it scoped.